crates/core/src/node/inner/node_executor.rs

```diff
@@ -263,19 +266,19 @@ impl NodeExecutor {
     async fn reset_fork_block_number(
         &mut self,
         block_number: L2BlockNumber,
-        reply: oneshot::Sender<anyhow::Result<()>>,
+        reply: oneshot::Sender<Result<(), AnvilNodeError>>,
     ) {
         let result = async {
             let node_inner = self.node_inner.write().await;
             let url = node_inner
                 .fork
                 .url()
-                .ok_or_else(|| anyhow::anyhow!("no existing fork found"))?;
+                .ok_or_else(|| anvil_zksync::node::generic_error!("no existing fork found"))?;
             // Keep scale factors as this is the same chain.
             let details = node_inner
                 .fork
                 .details()
-                .ok_or_else(|| anyhow::anyhow!("no existing fork found"))?;
+                .ok_or_else(|| anvil_zksync::node::generic_error!("no existing fork found"))?;
```

Comment:
This might make sense as an error kind but can be done in a follow-up PR

Reply:
Hm, when exactly does it trigger and what can user do about it?

Reply:
User is trying to manipulate/remove the fork but the node is not running in forking mode. User should make sure they either started in forking mode or switched through one of the available methods (`anvil_reset` is one, there might be others)